### DISCUSSION TITLE
Code view:  symbols panel is unhelpful and interferes with work

### DISCUSSION THREAD

**Discussion by dcupp** [community/community#54962] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

When I am viewing a code in Github's web UI, I used to be able to quickly jump between places in a file by double-clicking on a word (to quickly highlight it), and then quickly hitting cmd+c, cmd+f, cmd+v and enter, to copy and paste the symbol into the browsers search feature.

Now, half the time, when I double click on a word, this strange UI called "All symbols" appears on the right with a bunch of copies of that word, so using the browsers text search is useless, because there are a bunch of fake instances of the word.  It seems to be very hit or miss with symbol detection, and all it does it present a wall of appearances with very little context.

I used to use github's web UI because it would allow me to browse code "statically" without all of the heavyweight interference you get from a modern IDE.  It is very distracting to have the UI shift just because I double clicked on a word.  It would be great if:
1)  if there was a way to disable this.  Or, 
2) we could just hit ESC to make it go away
3) anyone can point me to a chrome/firefox plugin that can get rid of it

**Comment by claudio-luis** [comment]
+1

Was just now in the middle of reviewing a PR, and for one of the files I had the need to open the full file view, since it was easier to grasp the full context in this case, and I got unpleasantly surprised by this.

Not only does it actively make things much harder for me at least, but it has massive bugs that lead for the raw file to differ from what this new view is showing you.

In other words, I cannot even trust this new view to show me what the actual code looks like, because it's actively showing code that is actually not there.

As an example:
```php
try {
    //...
} catch (\Exception $exception) {
    //...
}
```
is actually showing as:
```php
try {
    //...
} catch (\Exception $ $exception) {
    //...
}
```
which is probably due to the backslash, and trying to select the text I see that there's a second layer of the same characters behind, so probably something is just causing a misalignment rather than messing with the code, but this didn't happen before with the previous view.

This new update should at least be given the option to be disabled, because it clearly isn't even stable enough for its view to be trusted yet, let alone the rest of the functionality it supposedly offers.

**Reply by adamlui** [reply]
You can (auto-)disable it with this userscript I wrote https://github.com/adamlui/github-widescreen

**Comment by JenieX** [comment]
I just don't think I'll ever get used to the new changes. 😞 

Please add an option to switch back to everything that was great before.

**Reply by adamlui** [reply]
I made a userscript that disables the side panel https://github.com/adamlui/github-widescreen

**Comment by dcupp** [comment]
This is also breaking a chrome extension for viewing encrypted ansible files in PRs:  https://github.com/gwatts/chrome-ansible/issues/7

**Reply by adamlui** [reply]
Check it out, I wrote a userscript to auto-hide it https://github.com/adamlui/github-widescreen

**Comment by cmcmA20** [comment]
This stuff doesn't respect even the lexing rules of my target language (Agda), not to mention scoping. In the current form the feature is not only unheplful, it's outright misleading.

You must disable this feature for files containing code in languages you don't have proper support of.

**Reply by adamlui** [reply]
You can use this userscript I wrote to auto-hide it https://github.com/adamlui/github-widescreen

**Comment by evdcush** [comment]
Please allow us to disable this feature.

- It's dead slow (always freezes for a few seconds as it apparently reloads the index, again)
- We have no control over its behavior
  - even if you close the UI, it will pop-up (and freeze) as soon as you click somewhere in the code

Please don't force opinionated features on users.

**Reply by aschwtzr** [reply]
I second this. Might be useful at some point but currently feels half-baked and distracting.

**Reply by dmwin72015** [reply]
> Please allow us to disable this feature.
> 
> * It's dead slow (always freezes for a few seconds as it apparently reloads the index, again)
> * We have no control over its behavior
>   
>   * even if you close the UI, it will pop-up (and freeze) as soon as you click somewhere in the code
> 
> Please don't force opinionated features on users.

Agree with you，also it makes me very angry.

**Reply by adamlui** [reply]
Guys I wrote a userscript that autohides it https://github.com/adamlui/github-widescreen

**Reply by michael-freidgeim-webjet** [reply]
@adamlui , does hiding help with slowness?
Can I apply user-script to Chrome on IPAD?

**Reply by adamlui** [reply]
Slowness of the new UI in general? Probably helps very slightly (since one less component to render).

IDK for iOS Chrome but Safari has https://github.com/quoid/userscripts

**Comment by mingyEx** [comment]
I support making this optional, because even in the IDE you don't open this panel by double-clicking on a line of code, it's just too obtrusive!

**Reply by adamlui** [reply]
You can make it unobtrusive with this userscript https://github.com/adamlui/github-widescreen

**Comment by chrisdahms-embark** [comment]
100% agree with the above that I find this new feature to be more a distraction than helpful and at a minimum this should be optional.  I have 2 questions at this time:

1) How can we notify somebody at GitHub (other than making this post, but I'm not sure this post alone will be motivation enough for whoever made this decision to change it).

2) Has anybody found a creative way to disable this feature?  I went through the account settings today and did not find an option to disable this.

**Reply by mingyEx** [reply]
Until an official solution to the problem is released
Maybe we can make a tampermonkey script or browser plugin to block mouse clicks on the code display area to temporarily satisfy the requirement.
But I don't know anything about this, anyone want to try it?

**Reply by JenieX** [reply]
Try this [user script](https://greasyfork.org/en/scripts/467622-jx-github-no-symbols-panel).

**Reply by adamlui** [reply]
![image](https://github.com/community/community/assets/10906554/a2ab23e9-3635-4060-9887-263ed20f91b9)

But here's an alternative one (I made) https://github.com/adamlui/github-widescreen

**Comment by mpiggott** [comment]
Its also distracting when clicking on part of the file and having text in the side panel changing.

**Reply by adamlui** [reply]
I made a userscript that disables the side panel https://github.com/adamlui/github-widescreen

**Comment by emteknetnz** [comment]
Please add an option to disable the feature entirely. I understand that some people will love this feature and I respect that. However many others will simply resent this feature. There is a long history of some people preferring lightweight text-editors and the github UI needs to allow for this.

**Reply by adamlui** [reply]
I made a userscript that disables the side panel https://github.com/adamlui/github-widescreen

**Comment by guzba** [comment]
You can block the symbols pane from messing with your UI with the following uBlock snippet:

`github.com###symbols-pane:upward(1)`

**Reply by adamlui** [reply]
Also you could use this userscript https://github.com/adamlui/github-widescreen

**Comment by Xant1k** [comment]
Developers, u must disable this shit. Useless.

**Reply by adamlui** [reply]
Check it oust I have created a userscript that attempts to disable it https://github.com/adamlui/github-widescreen

**Comment by ghost** [comment]
OK it seems you can somewhat disable this now. click More file actions, then under View options disable Open symbols on click. Also, to disable the cursor, you can use this filter with uBlock Origin:

~~~
github.com##.code-navigation-cursor
~~~

or for another method, everything is disabled if you are logged out.

**Reply by dcupp** [reply]
Thanks for sharing this.  I just confirmed it works:    [ ... ] -> [ Open Symbols on Click ]

**Reply by manaskarekar** [reply]
Apologies for missing the obvious, but where is this "More file actions"?

**Reply by dms117** [reply]
Agreed, I have no idea how to get to the settings described, if you could share that would be appreciated.

**Reply by WhiteyDude** [reply]
@manaskarekar and @dms117 there's a "..." button above "History", hidden if you're even slightly scrolled down the page. Within this menu is a tick menu item for "Open Symbols on Click" which thankfully unticking disables the popup.

**Reply by dms117** [reply]
Thanks @WhiteyDude I appreciate the reply.  I did eventually get to this option, the disconnect I was having was not realizing that I had to be looking at code in GitHub in order to have that button available - I was trying to find it while I was in a folder view, and so that's where my confusion arose.  Once I opened a file I saw the "..." right away and was able to adjust the setting.

**Reply by manaskarekar** [reply]
@WhiteyDude That works! 

@dms117 I was looking for it at the wrong place just like you were!

**Reply by AaronCloudZero** [reply]
Thanks to everyone who explained how to find this option, I would have _never_ found it on my own. That symbols panel makes viewing code on GitHub an unpleasant experience.

**Reply by harshblog150** [reply]
Thanks @WhiteyDude, & Everyone  , worked!

**Reply by adamlui** [reply]
I wrote this userscript that auto-disables left- and right-hand sidebars on first loads (can easily click to show them if still wanted) https://github.com/adamlui/github-widescreen

**Reply by ldcMasa** [reply]
I don't see any "More file actions" that I could click.

**Reply by ivanvoid** [reply]
Click 3 point menu button in a right top corner " . . . " and de-select "Open symbols on click". If you hover on it it will say "More file actions".
@ldcMasa

**Comment by Sleepful** [comment]
This is terrible in terms of accessibility, I like to use really large fonts on my computer to be able to read the text, so I zoom-in on the page.

Well now I cannot select any text while zoomed-in because the awful panel pops and drags my zoom all the way to the side of the page and stops me from reading the actual code.

**Reply by manaskarekar** [reply]
Not just that, it unselects the text if you close the dialog.

**Reply by adamlui** [reply]
You can auto-hide it on first load with this userscript I wrote https://github.com/adamlui/github-widescreen

**Comment by stijnh** [comment]
Is there already a way to disable the symbols panel yet? The alignment is incorrect in firefox and there are strange "phantom" symbols behind the actual code that mess up word search

**Reply by adamlui** [reply]
Yes you can install this userscript https://github.com/adamlui/github-widescreen

**Comment by wangchou** [comment]
I think a better approach on UX is **memorizing user preference**. If I clicked "hide symbols panel" button, GitHub should not show it next time until I open it again.

PS: I always close it either on 13" laptop or 27" 4k external display.

**Reply by adamlui** [reply]
You could install this userscript on those systems, it auto-hides it on first load https://github.com/adamlui/github-widescreen

**Comment by SGrondin** [comment]
It's really distracting. I agree that it should open on demand, not by default.

**Reply by adamlui** [reply]
Check it out, I wrote this script that hides it https://github.com/adamlui/github-widescreen

**Comment by adamlui** [comment]
I wrote a userscript that auto-hides it on first load (also left-hand sidebars too) https://github.com/adamlui/github-widescreen

**Reply by frobnitzem** [reply]
We get it.  Stop spamming this thread.

**Comment by michael-freidgeim-webjet** [comment]
As a workaround you can use **Raw** button
From https://github.com/orgs/community/discussions/23282#discussioncomment-3239808

**Comment by visortelle** [comment]
As a workaround, you can use the https://www.tampermonkey.net/ with this snippet:

```js
// ==UserScript==
// @name         Hide GitHub Symbols sidebar
// @namespace    http://tampermonkey.net/
// @version      2023-12-06
// @description  try to take over the world!
// @author       You
// @match        https://github.com/*
// @icon         data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==
// @grant        GM_addStyle
// ==/UserScript==

(function() {
    'use strict';

    GM_addStyle('div:has(> #symbols-pane) { display: none; }');
})();
```

**Comment by duduindo** [comment]
Filter with uBlock Origin:

```plain
github.com##.code-navigation-cursor
github.com###symbols-pane:upward(1)
github.com##.lbQLdc
```

**Reply by ugwuchijioke** [reply]
v1.1.0-IT-422-PR-12+BB-develop

**Reply by fadhil-riyanto** [reply]
Thanks, its works!

**Comment by Hanrryshani** [comment]
That new All symbols panel in GitHub is getting in the way, right. Double-clicking just to search in my browser is a mess now.
Here's what we can do,Click n' flee: Double-click, then click outside the code area before the panel pops up. Bam, gone!
Keyboard ninja: Skip the double-click. Use Ctrl+F (Windows/Linux) or Cmd+F (Mac) to open browser search, then Ctrl/Cmd+G to jump to the word. You're a search master.
Future hope: Keep an eye on GitHub settings, they might add a way to ditch the panel someday. Fingers crossed

**Reply by studokim** [reply]
They already have made that setting:
![image](https://github.com/community/community/assets/48129887/3d0e29f6-6a58-4241-8660-ac30a8ef22be)

**Reply by michael-freidgeim-webjet** [reply]
The option “open symbols on click” doest’t work on my iPad Safari. I’ve changed it to unticked, but symbols panel still is shown

**Reply by RafalSkolasinski** [reply]
Thanks, this is exactly the solution I needed. Ublock rules still leave the pieces of code highlighted in yellow.

**Comment by nmtitov** [comment]
I'd be happy to disable this panel because I don't use it and it opens when I double click on lines of code

**Comment by etherwitch** [comment]
Here's the deal - squeaky wheels get the grease. I don't mind that this feature exists but when I close the side-bar it shouldn't open itself back up the next time I highlight something. I have to highlight text to read it. Having this feature insist on re-opening itself makes it so difficult work. Being able to disable this is an accessibility requirement.

Thanks to the folks who contributed the uBlock snippits, that has been so helpful.

**Comment by shukriadams** [comment]
+1 to giving us the option to disable this.

**Comment by atemate** [comment]
+1!

**Reply by RemiCardona** [reply, marked as answer]
it's available in the "3 dots" menu next to the "go to file" search bar. You need to scroll back up to the top of the page to see that menu button. It's so unintuitive to get to that option, but it does exist.

Imho that option should be in or close to the "<>" button that opens the symbol viewer.

**Reply by shukriadams** [reply]
Thanks - that actually worked! Also, for anyone else reading this, reload the page for changes to take effect. And yes, why didn't Github put the option to disable the symbol viewer, inside the symbol viewer?

**Reply by bryankaplan** [reply]
Yes, but double-click to select is still broken, even with the symbols panel disabled. There's clearly an event handler intercepting it.

**Reply by duduindo** [reply]
Thank you!!!!

**Reply by sethrh** [reply]
This is better, thanks. I wish it were a full a solution.

Now I've got "CODE VIEW IS READ ONLY" following me around because the files are in some kind of read-only text editor. Typing activates the search box and still pops up the symbols panel. The code behind this thing seems to be buggy.

There should be an option to disable this whole thing across the site.

If they're trying to make a full web IDE, fine. That's sometimes useful. Give me a button that says "open this in the Web IDE" like Gitlab does, and leave the rest of the stuff plain text.

**Reply by su322** [reply]
GREAT!

**Comment by 9mm** [comment]
I **hate** this thing. Seething hatred. Why would you ever interfere with the ability to double-click copy. This is one of the most frustrating unhelpful features I've ever seen.

**Reply by duduindo** [reply]
That's true :joy:

**Comment by skozin** [comment]
Disabling the Symbols panel still doesn't fix the selection on double click: it keeps randomly failing. Even if the symbols search actually worked (it doesn't, at least for the Solidity code), it would not be worth breaking text selection which is the most basic function of any text viewer or editor. Please fix it.

**Comment by kilgarenone** [comment]
I have disabled everything under 'View options' but still my cursor does not land where I click. Fundamental action like drag and select texts don't work either. 

I just want my codes to be boring plain-texts by default, and enable those fancy stuff if I choose so. 

I'm on Firefox and windows.

**Comment by WaterSwallow** [comment]
> ### Select Topic Area
> Product Feedback
> 
> ### Body
> When I am viewing a code in Github's web UI, I used to be able to quickly jump between places in a file by double-clicking on a word (to quickly highlight it), and then quickly hitting cmd+c, cmd+f, cmd+v and enter, to copy and paste the symbol into the browsers search feature.
> 
> Now, half the time, when I double click on a word, this strange UI called "All symbols" appears on the right with a bunch of copies of that word, so using the browsers text search is useless, because there are a bunch of fake instances of the word. It seems to be very hit or miss with symbol detection, and all it does it present a wall of appearances with very little context.
> 
> I used to use github's web UI because it would allow me to browse code "statically" without all of the heavyweight interference you get from a modern IDE. It is very distracting to have the UI shift just because I double clicked on a word. It would be great if:
> 
> 1. if there was a way to disable this.  Or,
> 2. we could just hit ESC to make it go away
> 3. anyone can point me to a chrome/firefox plugin that can get rid of it

Thanks for sharing your experience and feedback on Github's web UI shift upon double-clicking a word. 

Indeed, a sudden appearance of the "All symbols" panel could feel disruptive, especially if it interferes with your established workflow of substituting a full-fledged IDE in favor of a static way of browsing code on GitHub's Web UI. 

Your suggestions of potentially disabling this UI shift or adding a simple shortcut like ESC to close the panel seem very user-centered and pragmatic. I am not currently aware of any specific browser plugins designed to disable it, but there are developers continually creating extensions to enhance user experience across platforms, so it might be worth looking around.

Meanwhile, bringing issues like this to the fore helps GitHub understand user concerns and could direct their attention towards improving user experience based on the valuable feedback you provided. I would recommend sharing this feedback directly with GitHub as well.

**Comment by tgrushka** [comment]
Accessibility. It's a brand new concept to most devs, I guess. The popup is terrible for those of us who have to double-click-select and read with our screen readers.

BTW, the lack of ability to configure font size is also terrible. Doing it in CSS works a little, but very hacky, as they're also using a hidden text field to do selection.

Ugly interface. Billion dollar company, everyone depends on them, UX goes down the tubes. Typical.

**Comment by kostakis** [comment]
+1 This thing is not good and interferes with the actual work.
Even if you close it once will reopen on the next session. Wish if we ha a setting to permanently turn off.

**Comment by jtomkiew-mng** [comment]
Double click still sometimes gets captured by this "feature" and fails to select a word, lovely 🤗

**Comment by joshuataylor** [comment]
And the latest update has broken code view again, it doesn't highlight properly, it seems to think it's a symbol(?), and won't let me highlight the line properly. 🤦 

<img width="2395" height="1174" alt="image" src="https://github.com/user-attachments/assets/08069df5-f37a-4b52-b09b-60fd765776b7" />

Ridiculous.

**Comment by joshuataylor** [comment]
Now there's a cursor. Lots of changes recently to make it even worse.

uBlock Origin rules..

```
github.com##.react-code-file-contents span:style(background-color: transparent !important)
github.com##[data-testid="navigation-cursor"]
```
